transport/tchannel/config.go

```diff
@@ -61,7 +61,16 @@ type OutboundConfig struct {
 
 // TransportSpec returns a TransportSpec for the TChannel unary transport.
 func TransportSpec(opts ...Option) config.TransportSpec {
-	return (&transportSpec{}).Spec(opts...)
+	var ts transportSpec
+	for _, o := range opts {
+		switch opt := o.(type) {
+		case TransportOption:
+			ts.transportOptions = append(ts.transportOptions, opt)
+		default:
+			panic(fmt.Sprintf("unknown option of type %T: %v", o, o))
```

Comment:
Added a test. This wasn't tested before because it would be a test for the sake of coverage; this can't happen in real code because we control all instances of Option.